easybuild/easyblocks/x/xmipp.py

```diff
@@ -0,0 +1,116 @@
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Xmipp, implemented as an easyblock
+
+@author: Jens Timmerman (Ghent University)
+"""
+import glob
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.filetools import mkdir, extract_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+
+class EB_Xmipp(EasyBlock):
+    """Support for building/installing Xmipp."""
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, enable building in installation directory."""
+        super(self.__class__, self).__init__(*args, **kwargs)
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Extract sources."""
+        # strip off 'xmipp' part to avoid having everything in a 'xmipp' subdirectory
+        self.cfg['unpack_options'] = "--strip-components=1"
+        super(self.__class__, self).extract_step()
+
+    def configure_step(self):
+        """Configure xmipp build via a provided wrapper around sconss."""
+        # install step expects these to exist
+        mkdir('lib')
+        mkdir('bin')
+        # run interactive 'config' script to generate install.info file
+        try:
+            # ln -s `which python` xmipp_python
+            orig = os.path.join(get_software_root('Python'), 'bin', 'python')
+            dest = 'xmipp_python'
+            os.symlink(orig, dest)
+            # ln -s $EBROOTPYTHON/lib/python2.7 lib/python2.7
+            self.python_short_ver = ".".join(get_software_version('Python').split(".")[0:2])
+            orig = os.path.join(get_software_root('Python'), 'lib', 'python%s' % self.python_short_ver)
+            dest = os.path.join('lib', 'python%s' % self.python_short_ver)
+            os.symlink(orig, dest)
+        except OSError as err:
+            self.log.error("Failed to symlink %s to %s: %s" % (orig, dest, err))
```

Comment:
will this pick up the `orig` and `dest` from the `try` block?

I'd rather have *only* the call to `os.symlink` in a `try`-`except` block, with an error like:

```python
self.log.error("Failed to create symlinks: %s" % err)
```